fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/api/networking/v1/ClientNetworking.java

```diff
@@ -0,0 +1,151 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.networking.v1;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.GenericFutureListener;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.ClientLoginNetworkHandler;
+import net.minecraft.client.network.ClientPlayNetworkHandler;
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.network.PacketByteBuf;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.impl.networking.client.ClientNetworkingDetails;
+
+/**
+ * Offers access to client-side networking functionalities.
+ *
+ * <p>Client-side networking functionalities include receiving clientbound packets,
+ * sending serverbound packets, and events related to client-side network handlers.</p>
+ *
+ * <p>This class should be only used on the physical client and for the logical client.</p>
+ *
+ * @see ServerNetworking
+ */
+@Environment(EnvType.CLIENT)
+public final class ClientNetworking {
+	/**
+	 * Returns the packet sender for the current client player.
+	 *
+	 * <p>This is a shortcut method for getting a sender.
+	 * When a client play network handler is available, {@link #getPlaySender(ClientPlayNetworkHandler)} is preferred.
+	 *
+	 * @return the packet sender for the current client player
+	 * @throws IllegalStateException if the client's player is {@code null}
+	 */
+	public static PlayPacketSender getPlaySender() throws IllegalStateException {
+		ClientPlayerEntity player = MinecraftClient.getInstance().player;
+
+		if (player == null) {
+			throw new IllegalStateException("Cannot get packet sender when not in game!");
+		}
+
+		return getPlaySender(player.networkHandler);
+	}
+
+	/**
+	 * Returns the packet sender for a client play network handler.
+	 *
+	 * @param handler a client play network handler
+	 * @return the associated packet sender
+	 */
+	public static PlayPacketSender getPlaySender(ClientPlayNetworkHandler handler) {
+		return ClientNetworkingDetails.getAddon(handler);
+	}
+
+	/**
+	 * Returns the packet receiver for channel handler registration on client play network handlers, receiving {@link net.minecraft.network.packet.s2c.play.CustomPayloadS2CPacket server to client custom payload packets}.
+	 */
+	public static ChannelHandlerRegistry<PlayChannelHandler> getPlayReceiver() {
```

Comment:
I might change this to `getPlayReceivers` to clarify that this isn't returning one specific play receiver but a registry of play receivers